google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java

```diff
@@ -1077,9 +1077,9 @@ public void bulkMutateRows(BulkMutation mutation) {
   }
 
   /**
-   * Reads rows for given tableId in a batch. If the row does not exist, the value will be null.
-   * This operation should be called with in a single thread. The returned Batcher instance is not
-   * threadsafe, it can only be used from single thread.
+   * Reads rows for given tableId in a batch. If the row does not exist, the value will be null. The
+   * keys are processed sequentially one by one in the order they are passed to the Batcher. The
+   * returned Batcher instance is not threadsafe, it can only be used from a single thread.
```

Comment:
This isnt quite right.
The main issue is that Bigtable ReadRows must return row in lexicographic  row key order. Which means that the keys will be processed sequentially on the serverside. So batching works best when the caller cares more about throughput than latency and the keys are grouped together.

If the caller cares more about latency then sending smaller concurrent request is better

Reply:
Thanks for the clarification, now I see what this issue is about.
Please take a look at the updated variant.